File: Dev/C++/Mascaret/src/BEHAVE/Mailbox.cpp
#include "BEHAVE/Mailbox.h"
#include "BEHAVE/ACLMessage.h"

namespace BEHAVE {

Mailbox::Mailbox()
{}

Mailbox::~Mailbox()
{}

shared_ptr<ACLMessage> Mailbox::check()
{
	shared_ptr<ACLMessage> msg;
	//if (!_messagesQueue.isEmpty())
	if (_messagesQueue.size())
	{
		msg = _messagesQueue[0];
		_messagesQueue.erase(_messagesQueue.begin());
		_messagesChecked.push_back(msg);
	}
	return msg;

}

}

File: Dev/CS/Mascaret/Mascaret/BEHAVE/Mailbox.cs
using System;
using System.Collections.Generic;

namespace Mascaret
{
    public class Mailbox
    {

        private List<ACLMessage> messagesQueue = new List<ACLMessage>();
        public List<ACLMessage> MessagesQueue
        {
            get { return messagesQueue; }
            set { messagesQueue = value; }
        }


        private List<ACLMessage> messagesChecked = new List<ACLMessage>();
        public List<ACLMessage> MessagesChecked
        {
            get { return messagesChecked; }
            set { messagesChecked = value; }
        }


        private List<ACLMessage> messagesSent = new List<ACLMessage>();
        public List<ACLMessage> MessagesSent
        {
            get { return messagesSent; }
            set { messagesSent = value; }
        }


        public void postMessage(ACLMessage message)
        {
            messagesQueue.Add(message);
        }

        public ACLMessage check()
        {
            ACLMessage msg = null;

            if (messagesQueue.Count != 0)
            {
                msg = messagesQueue[0];
                messagesQueue.RemoveRange(0, 1);
                messagesChecked.Add(msg);
            }

            return msg;
        }

        public void send(ACLMessage message)
        {
            messagesSent.Add(message);
        }
    }
}

